halo2/src/circuit_builder.rs

```diff
@@ -171,6 +176,115 @@ pub(crate) fn analyzed_to_circuit<T: FieldElement>(
         }
     }
 
+    if lookups.is_empty() {
+        // TODO something inside halo2 breaks (only in debug mode) if lookups is empty,
+        // so just add an empty lookup.
+        lookups.push(Lookup {
+            name: "".to_string(),
+            exps: (vec![], vec![]),
+        });
+    }
+
+    // build Plaf fixed. -------------------------------------------------------------------------
+
+    let fixed: Vec<Vec<_>> = cd
+        .fixed
+        .iter()
+        .map(|(_, row)| {
+            row.iter()
+                .map(|value| Some(value.to_arbitrary_integer()))
+                .collect()
+        })
+        .collect();
+
+    let mut copies = vec![];
+    let mut publics = vec![];
```

Comment:
This is is unused.

Also, this block seems to be copied, maybe we can extract a function?

Reply:
After simplifying this the `analyzed_to_*` API also looks much better!